server/src/main/java/io/crate/replication/logical/LogicalReplicationService.java

```diff
@@ -319,6 +309,9 @@ public CompletableFuture<Boolean> restore(String subscriptionName,
 
         FutureActionListener<RestoreService.RestoreCompletionResponse> restoreListener = new FutureActionListener<>();
         activeOperations.incrementAndGet();
+        Set<RelationName> relationNames = tablesToRestore.stream()
+            .map(TableOrPartition::table)
+            .collect(Collectors.toUnmodifiableSet());
```

Comment:
Do  we do it differently from `RestoreSnapshotPlan` and don't handle paritionIdent due to https://github.com/crate/crate/issues/12451? Maybe worth adding a short comment

Reply:
No, the relationNames here are used to update the subscription states to `restoring`. For the recovery itself the list of TableOrPartition is used.

The logical replication needs to create new partitions for partitioned tables, and that's working on index level. The linked issue is about limiting replication to selected partitions instead.